drivers/sensor/bosch/bmi08x/bmi08x_accel.c

```diff
@@ -454,19 +454,31 @@ static int bmi08x_temp_channel_get(const struct device *dev, struct sensor_value
 {
 	uint16_t temp_raw = 0U;
 	int32_t temp_micro = 0;
+	int16_t temp_int11 = 0;
 	int ret;
 
 	ret = bmi08x_accel_word_read(dev, BMI08X_REG_TEMP_MSB, &temp_raw);
-	if (ret < 0) {
-		return ret;
-	}
+	temp_int11 = (temp_raw & 0xFF) << 3;
 
-	/* the scale is 1/2^5/LSB = 31250 micro degrees */
-	temp_micro = BMI08X_TEMP_OFFSET * 1000000ULL + temp_raw * 31250ULL;
+	if (temp_raw == 0x80) {
+		/* temperature invalid */
+		return -ENODATA;
+	}
 
+	ret |= bmi08x_accel_word_read(dev, BMI08X_REG_TEMP_LSB, &temp_raw);
```

Comment:
You might not have a valid errno after the bitwise OR operation

Reply:
Corrected